### DISCUSSION TITLE
How this lib works with ref and onChange?

### DISCUSSION THREAD

**Discussion by pedrobslisboa** [uNmAnNeR/imaskjs#824] [Q&A]
I've tried to replicate it many times and failed on every single one. Already read the code and even tried chatgpt for help and couldn't do it. What is the trick?

How do the onChange trigger with the ref option provided? 
Notice, the ref is used to change the value, we don't have the value attr used on the input elem.
```js
function IMaskWithHook () {
  const [ opts, setOpts ] = useState({ mask: Number });
  const {
    ref,
    maskRef,
    value,
    setValue,
    unmaskedValue,
    setUnmaskedValue,
    typedValue,
    setTypedValue,
  } = useIMask(opts);
  
  return (
    <input ref={ref} onChange={(e) => console.log(e.target.value)} />
  );
}
```

Every time I try to reproduce it, the onChange doesn't trigger.

Because this kind of feature can be used in many ways, it would be nice to know how to achieve it.

Thank you all :3

**Comment by rafde** [comment]
You have to use onAccept to see the new value
```jsx
import { useState } from 'react';
import { useIMask } from 'react-imask';

function IMaskWithHook () {
  const [ opts, setOpts ] = useState({ mask: Number });
  const {
    ref,
    maskRef,
    value,
    setValue,
    unmaskedValue,
    setUnmaskedValue,
    typedValue,
    setTypedValue,
  } = useIMask(opts, { onAccept(value) { console.log(value); }, });
  
  return (
    <input ref={ref} />
  );
}
```
